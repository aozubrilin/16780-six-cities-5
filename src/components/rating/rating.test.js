import React from "react";
import renderer from "react-test-renderer";
import {configure, mount} from "enzyme";
import Rating from "./rating";
import Adapter from "enzyme-adapter-react-16";

configure({adapter: new Adapter()});
const noop = () => {};
describe(`Rating snapshot`, () => {
  it(`Should Rating render correctly`, () => {
    const tree = renderer.create(
        <Rating
          rating={`4`}
          onChange={noop}
        />
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Rating checked  3 stars`, () => {
    const tree = mount(
        <Rating
          rating={`3`}
          offerId={`1`}
          onChange={noop}
        />
    );

    const starsInput = tree.find(`.form__rating-input`).at(3);
    starsInput.simulate(`change`);
    expect(tree.debug()).toMatchSnapshot();
  });

});
