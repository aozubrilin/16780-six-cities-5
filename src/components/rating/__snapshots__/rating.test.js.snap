// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating snapshot Rating checked  3 stars 1`] = `
"<Rating rating=\\"3\\" offerId=\\"1\\" onChange={[Function: noop]}>
  <div className=\\"reviews__rating-form form__rating\\">
    <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"5\\" id=\\"5-stars\\" type=\\"radio\\" onChange={[Function: noop]} checked={false} />
    <label htmlFor=\\"5-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\">
      <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
        <use xlinkHref=\\"#icon-star\\" />
      </svg>
    </label>
    <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"4\\" id=\\"4-stars\\" type=\\"radio\\" onChange={[Function: noop]} checked={false} />
    <label htmlFor=\\"4-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"good\\">
      <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
        <use xlinkHref=\\"#icon-star\\" />
      </svg>
    </label>
    <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"3\\" id=\\"3-stars\\" type=\\"radio\\" onChange={[Function: noop]} checked={true} />
    <label htmlFor=\\"3-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"not bad\\">
      <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
        <use xlinkHref=\\"#icon-star\\" />
      </svg>
    </label>
    <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"2\\" id=\\"2-stars\\" type=\\"radio\\" onChange={[Function: noop]} checked={false} />
    <label htmlFor=\\"2-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"badly\\">
      <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
        <use xlinkHref=\\"#icon-star\\" />
      </svg>
    </label>
    <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"1\\" id=\\"1-stars\\" type=\\"radio\\" onChange={[Function: noop]} checked={false} />
    <label htmlFor=\\"1-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"terribly\\">
      <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
        <use xlinkHref=\\"#icon-star\\" />
      </svg>
    </label>
  </div>
</Rating>"
`;

exports[`Rating snapshot Should Rating render correctly 1`] = `
<div
  className="reviews__rating-form form__rating"
>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    id="5-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value="5"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={true}
    className="form__rating-input visually-hidden"
    id="4-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value="4"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="4-stars"
    title="good"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    id="3-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value="3"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="3-stars"
    title="not bad"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    id="2-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value="2"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="2-stars"
    title="badly"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    id="1-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value="1"
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="1-stars"
    title="terribly"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
</div>
`;
